src/int_allocator.rs

```diff
@@ -70,6 +70,28 @@ impl Allocator for IntAllocator {
         r
     }
 
+    fn new_substr(&mut self, node: Self::Ptr, start: u32, end: u32) -> Self::Ptr {
+        if node >= 0 {
```

Comment:
Maybe just wrap in `Result` rather than panic.

Reply:
Yes, my next patch introduces the ability for allocations to fail, I'm fixing this there